app/scripts/contentscript.js

```diff
@@ -103,6 +104,23 @@ const IGNORE_INIT_METHODS_FOR_KEEP_ALIVE = [
 let keepAliveInterval;
 let keepAliveTimer;
 
+/**
+ * Sending a message to the extension to receive will keep the service worker alive.
+ *
+ * If the extension is unloaded or reloaded during a session and the user attempts to send a
+ * message to the extension, an "Extension context invalidated." error will occur.
+ * When this happens, prompt the user to reload the extension.
+ */
+const sendMessageWorkerKeepAlive = () => {
+  browser.runtime
+    .sendMessage({ name: WORKER_KEEP_ALIVE_MESSAGE })
+    .then(null, (error) => {
+      error.message === 'Extension context invalidated.'
```

Comment:
Is this message stable across browsers and languages?

Reply:
@legobeat great question! thanks for asking 

It appears so. 

It seems that chromium has the error message fixed to 'Extension context invalidated.' and does not send this error in other languages: https://github.com/chromium/chromium/blob/main/extensions/renderer/gin_port.cc#L31

Before checking the chromium repo, I did verify that the error remains the same here and on Opera Beta with Spanish as the default language
<img width="1173" alt="Screenshot 2022-10-31 at 8 54 51 PM" src="https://user-images.githubusercontent.com/20778143/199032048-317c3f2d-5549-47ad-b400-e1a2e90140ca.png">

I went with checking the direct string because there is no error code provided with this error.

Reply:
is there an equivalent error on firefox?

Reply:
I'm receiving a different warning/error when running these repro steps in firefox. Needs investigation 

<img width="900" alt="Screen Shot 2022-11-22 at 17 30 10" src="https://user-images.githubusercontent.com/20778143/203629737-c21e62d3-5540-4ed0-b728-f5e32e5e784b.png">

Reply:
Thanks for the callouts!

Reply:
OK - is there a ticket for tracking FF support (perhaps alongside other FF MV3 items)? Just so it's followed up on appropriately.

Reply:
@legobeat Ok, I created a ticket here: https://github.com/MetaMask/metamask-extension/issues/16909. Thank you 🙏